src/node/mod.rs

```diff
@@ -155,16 +155,12 @@ impl<R: CryptoRng + Rng> Node<R> {
         while let Some(op) = next_op {
             next_op = match op {
                 Single(operation) => self.process(operation).await,
-                Multiple(ops) => {
-                    let mut node_op = None;
-                    for c in ops.into_iter() {
-                        if let Some(op) = self.process(c).await {
-                            node_op = op;
-                            break;
-                        }
-                    }
-                    Some(node_op)
-                }
+                Multiple(ops) => Some(
+                    ops.into_iter()
+                        .filter_map(|c| futures::executor::block_on(self.process(c)))
+                        .collect::<Vec<_>>()
+                        .into(),
+                ),
```

Comment:
We are avoiding the use of `futures::executor::block_on()`. 
Does this make sense?
```suggestion
                    let mut node_ops = Vec::new();
                    for c in ops.into_iter() {
                        if let Some(op) = self.process(c).await {
                            node_ops.push(op);
                        }
                    }
                    Some(node_ops)
```

Reply:
Yeah sure, absolutely. I agree blocking on the tasks is not desired, was just looking for a simple fix.

Reply:
Woops, tried to use the commit suggestion feature. Will force push.